test-app/tests/zoetrope/rendering-test.gts

```diff
@@ -0,0 +1,201 @@
+  });
+
+  test('header is visible if provided', async function (assert) {
+    await render(
+      <template>
+        <Zoetrope @offset={{40}}>
+          <:header>
+            <h1>Header</h1>
+          </:header>
+          <:content>
+            <a href="#">Card</a>
+            <a href="#">Card</a>
+            <a href="#">Card</a>
+          </:content>
+        </Zoetrope>
+      </template>
+    );
+
+    assert.dom('.zoetrope-header').hasText('Header');
+  });
+
+  test('no controls visible if no need to scroll', async function (assert) {
+    await render(
+      <template>
+        <Zoetrope @offset={{40}}>
+          <:header>
+            <h1>Header</h1>
+          </:header>
+          <:content>
+            <a href="#">Card</a>
+            <a href="#">Card</a>
+            <a href="#">Card</a>
+          </:content>
+        </Zoetrope>
+      </template>
+    );
+
+    await Promise.resolve();
+
+    assert.dom('.zoetrope-controls').doesNotExist();
+  });
+
+  test('controls visible if scrolling required', async function (assert) {
+    await render(
+      <template>
+        <Zoetrope @offset={{40}}>
+          <:header>
+            <h1>Header</h1>
+          </:header>
+          <:content>
+            <a href="#">Card</a>
+            <a href="#">Card</a>
+            <a href="#">Card</a>
+          </:content>
+        </Zoetrope>
+
+        {{! template-lint-disable no-forbidden-elements}}
+        <style>
+          a {width: 100%; display: block;}
+        </style>
+      </template>
+    );
+
+    assert.dom('.zoetrope-controls').exists();
+  });
+
+  test('can scroll right', async function (assert) {
+    await render(
+      <template>
+        {{! template-lint-disable no-forbidden-elements}}
+        <style>
+          .zoetrope {width: 400px;} a {width: 200px; display: block;}
+        </style>
+
+        <Zoetrope @scrollBehavior="instant">
+          <:header>
+            <h1>Header</h1>
+          </:header>
+          <:content>
+            <a href="#">Card</a>
+            <a href="#">Card</a>
+            <a href="#">Card</a>
+          </:content>
+        </Zoetrope>
+      </template>
+    );
+
+    await click('.zoetrope-controls button:last-child');
+
+    assert.dom('.zoetrope-scroller').hasProperty('scrollLeft', 208);
+  });
+
+  test('can scroll left', async function (assert) {
+    await render(
+      <template>
+        {{! template-lint-disable no-forbidden-elements}}
+        <style>
+          .zoetrope {width: 400px;} a {width: 200px; display: block;}
+        </style>
+
+        <Zoetrope @scrollBehavior="instant">
+          <:header>
+            <h1>Header</h1>
+          </:header>
+          <:content>
+            <a href="#">Card</a>
+            <a href="#">Card</a>
+            <a href="#">Card</a>
+          </:content>
+        </Zoetrope>
+      </template>
+    );
+
+    // scroll right first so we can scroll left
+    await click('.zoetrope-controls button:last-child');
```

Comment:
should probably make clicks tie in to settled state so that the await promise isn't needed.

when consumers of this component want to test, we don't want them to have to worry about the details of animation / timing / etc

Reply:
I see we use `scrollBy` -- so I think all we need to do add waitForPromise + new Promise + requestAnimationFrame

Reply:
With the new test waiter this RAF delay doesn't seem required (at least, locally it works without it)

Reply:
ye, I forgot to clean this up :see_no_evil: